framework/common/CMakeLists.txt

```diff
@@ -46,7 +46,7 @@ endif()
 
 # Create a single common library
 add_library(common ${sources})
-set_target_properties(common PROPERTIES PREFIX "mapf" VERSION ${prplmesh_VERSION} SOVERSION ${prplmesh_VERSION_MAJOR})
+set_target_properties(common PROPERTIES PREFIX "libmapf" VERSION ${prplmesh_VERSION} SOVERSION ${prplmesh_VERSION_MAJOR})
```

Comment:
There are [more options](https://stackoverflow.com/questions/31038963/how-do-you-rename-a-library-filename-in-cmake) to get the desired result.
Also note that since [PREFIX](https://cmake.org/cmake/help/latest/prop_tgt/PREFIX.html) property is used, the automated 'lib' prefix is removed, that is the purpose of setting the PREFIX property.

Reply:
In other words, your proposal is to just remove the PREFIX, right? I like that...

Reply:
CMake is _designed_ to set the prefix based on the target, so if we don't have special case we can leave it to CMake.
However, here it seems that from some reason the name `libcommon` was not enough, the required name is `libmapfcommon`. I don't know why and what were the considerations, but just removing the prefix will leave us with `libcommon`.

Reply:
change library name from 'common' to 'mapfcommon'